Tests/SnapshotTests.swift

```diff
@@ -28,4 +28,31 @@ final class SnapshotTests: XCTestCase {
             )
         )
     }
+    
+    func testTakeSnapshotWithPreprocessor() throws {
+        guard let directory = ProcessInfo.processInfo.environment["SNAPSHOT_DIR"] else {
+            fatalError("Set directory to the build environment variables with key `SNAPSHOT_DIR`.")
+        }
+
+        Playbook.default.add(AllScenarios.self)
+        
+        let preprocessor: ((UIView) -> UIView) = { view in
+            view.backgroundColor = .lightGray
+            return view
+        }
+
+        try Playbook.default.run(
+            Snapshot(
+                directory: URL(fileURLWithPath: directory),
```

Comment:
This test clears all snapshot taken by `testTakeSnapshot`, so you should separate the directory.

```swift
directory: URL(fileURLWithPath: directory).appendingPathComponent("GrayBackground")
```

FYI, visual regression testing is not working on CI because the report hosting server is down now. (I used GCS that paid by my former company) So, I'm gonna use [Percy](https://percy.io/) for regression testing.